### PR TITLE
Update the animation-worklet IDL file

### PR DESCRIPTION
**Pull request by lukebjerring**
Hello reviewer(s),

This PR is intended to consolidate the spec’s IDL definition with the WPT test suite’s copy, and any idlharness tests.

The up-to-date copy of the IDL file was automatically extracted from the reffy-reports repo (https://github.com/tidoust/reffy-reports/tree/master/whatwg/idl) which scrapes known specs automatically + regulary.

This PR is part of a migration project which will eventually be automatically updating and creating PRs for changes in spec IDL.

Please check that:
The Spec (and its source) is correct and up-to-date
All tests which cover the IDL in the spec have been migrated to fetch + use the idl in the `interfaces/` directory (instead of inline copies in the test files).

### CODE REVIEW

**Comment by w3c-bots** [comment]
# Build PASSED

Started: 2018-03-02 20:51:23
Finished: 2018-03-02 21:01:33


<h2>Failing Jobs</h2>
<ul>

<li>chrome:dev</li>

<li>safari:11.0</li>

<li>MicrosoftEdge:16.16299</li>

</ul>




View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/25164)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/348381384)

File: interfaces/animation-worklet.idl [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+};
+
+callback VoidFunction = void ();
+
+[Global=Worklet]
+interface AnimationWorkletGlobalScope : WorkletGlobalScope {
+    void registerAnimator(DOMString name, VoidFunction animatorCtor);
+};
+
+[Constructor (DOMString animatorName,
```
Comment (foolip) [inline interfaces/animation-worklet.idl]: Found a spec bug here: https://github.com/WICG/animation-worklet/issues/96

**Comment by foolip** [comment]
I've added a test but it's broken because of https://github.com/WICG/animation-worklet/issues/96, suggest just waiting until that's fixed.

**Comment by foolip** [comment]
@flackr @majido @stephenmcgruer this adds you as owners of this directory with a single test as well. Is that OK?

Maybe you want to put some tests in here? :) Tentative tests are fine: http://web-platform-tests.org/writing-tests/file-names.html

**Comment by majido** [comment]
@foolip adding us as owner sounds good.

I didn't know about `tentative`. We have a bunch of test for AnimationWorklet and ScrollTimeline that should be possible to upstream.

**Comment by foolip** [comment]
OK, https://github.com/WICG/animation-worklet/issues/96 is fixed, now this just needs to be synced with the new IDL.

**Comment by lukebjerring** [comment]
@foolip - updated, but the test is broken because of a bunch of undeclared deps (which aren't in any of the imported IDL).

**Comment by foolip** [comment]
Added a missing dep, and now it's broken because 'promise_test: Unhandled rejection with value: object "Partial interface AnimationEffectReadOnly with no original interface"'

AnimationEffectReadOnly used to be in https://drafts.csswg.org/web-animations/ but now there's just AnimationEffect. Tried to fix it but it was too messy, filed https://github.com/WICG/animation-worklet/issues/101.

**Review by foolip — APPROVED** [review]

**Pull request merged**